### ISSUE TITLE
Problem matching response header "HTTP/2 200 \r\n"

### ISSUE THREAD

**Issue by polaris64** [alphapapa/plz.el#12]
With `plz` after commit 74536c5396abe6be1691193dc3c816a2a73d4655 I no longer seem to be able to make requests to certain sites.

For example: -

```
(plz "GET" "https://blog.polaris64.net/")
```

signals the error `forward-sexp: HTTP error: "plz--response: Unable to parse HTTP response status line", "HTTP/2 200 ^M"`.

I can see that `plz-http-response-status-line-regexp` was changed [here](https://github.com/alphapapa/plz.el/commit/74536c5396abe6be1691193dc3c816a2a73d4655#diff-86628899659777b5eeb393a8c5c3471f4bdbe10ed16a653a12705cee1225d07dR113). Restoring the previous regexp allows the request to work successfully, returning the response as expected.

**Comment by polaris64** [comment]
I've just been experimenting with the regexp and this passes for this particular case: -

```elisp
(let ((http-res "HTTP/2 200 \n"))
  (string-match (rx bol "HTTP/" (group (1+ (or digit ".")))
                    ;; Status code
                    (1+ blank) (group (1+ digit))
                    ;; Reason phrase
                    (optional (1+ blank) (group (* (not (or "\r" "\n")))))
                    (or "\r" "\n" eol))
                http-res)
  (list (cons 'match (match-string 0 http-res))
        (cons 'version (match-string 1 http-res))
        (cons 'status (match-string 2 http-res))
        (cons 'reason (match-string 3 http-res))))
```
Result is: -

```
((match . "HTTP/2 200 
") (version . "2") (status . "200") (reason . ""))
```

Of course it needs testing in different situations but hopefully that helps.

**Comment by dannyob** [comment]
I'm seeing the same thing (including from https://httpbin.org/post )

**Comment by alphapapa** [comment]
@polaris64 @dannyob  Thanks.  This should now be fixed in v0.3.  Please let me know if it works for you.

**Comment by polaris64** [comment]
Hi @alphapapa, yes I can confirm that this works now. Thanks for fixing it!
